### PR TITLE
feat(Session Replay): Prepare Session event

### PR DESCRIPTION
**Pull request by brustolin**
Handling session event during `prepareEvent`.

_#skip-changelog_

### CODE REVIEW

File: Sources/Sentry/SentryScope.m [hunk 1]
```diff
@@ -496,7 +496,7 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
         event.fingerprint = fingerprints;
     }
 
-    if (event.breadcrumbs == nil) {
+    if (event.breadcrumbs == nil && ![event.type isEqualToString:SentryEnvelopeItemTypeReplayVideo]) {
```
Comment (brustolin) [inline Sources/Sentry/SentryScope.m]: @romtsn, @bruno-garcia
Apart from breadcrumb, should we clean replay event of any other data that usually goes with events?

Comment (romtsn) [reply Sources/Sentry/SentryScope.m]: I dunno if you set these ones right now, but I'm clearing them out:
```
replayEvent.extra = nil;
replayEvent.threads = nil;
replayEvent.debugMeta = nil;
```

Comment (brustolin) [reply Sources/Sentry/SentryScope.m]: Perfect! I will just save some cycles and not even copy them to the event.

**Review by philipphofmann — APPROVED** [review]
Some tests for this would be nice. If they already exist, LTGM. Otherwise, please add some.

**Pull request merged**